docs: highlight Haystack 3.0 features in README (#12085)
deepset-ai/haystack: README.md

## Features

**Agents built for production**  

> Extend agent behavior with lifecycle hooks (`before_llm`, `before_tool`, `on_exit`, …) for guardrails and custom logic, and track `step_count`, `token_usage`, and tool calls out of the box for monitoring and cost control.

Reviewer: We should also add pre-built agents and skills here or as another item on the list

**Built for context engineering**  
Design flexible systems with explicit control over how information is retrieved, ranked, filtered, combined, structured, and routed before it reaches the model. Define pipelines and agent workflows where retrieval, memory, tools, and generation are transparent and traceable.

**Unified pipelines, sync and async**  

> One `Pipeline` runs synchronously or asynchronously and streams token by token — no separate `AsyncPipeline`, no rewrites to switch execution modes.

Reviewer: Maybe we can focus on just having one pipeline now rather than explaining the difference. 

```suggestion
**Native Async Support**  
One `Pipeline` runs synchronously or asynchronously and streams token by token. `Agent` can run concurrent tool calls.  
```

**Modular and customizable**  
Use built-in components for retrieval, indexing, tool calling, memory, and evaluation, or create your own. Add loops, branches, and conditional logic to precisely control how context moves through your pipelines and agent workflows.

**Model- and vendor-agnostic**  

> Integrate with OpenAI, Mistral, Anthropic, Cohere, Hugging Face, Azure OpenAI, AWS Bedrock, local models, and many others. Swap models or infrastructure components without rewriting your system.

Reviewer: We can add Google to this list

## Features

**Agents built for production**  
Extend agent behavior with lifecycle hooks (`before_llm`, `before_tool`, `on_exit`, …) for guardrails and custom logic, and track `step_count`, `token_usage`, and tool calls out of the box for monitoring and cost control. Get started fast with ready-made agents from [Agent Pack](https://github.com/deepset-ai/haystack-core-integrations/tree/main/integrations/agent_pack) (e.g., a deep research agent, or an advanced RAG agent) or give your own agents progressive skill discovery via `SkillToolset`, so skill descriptions only enter context when needed.

**Built for context engineering**  
Design flexible systems with explicit control over how information is retrieved, ranked, filtered, combined, structured, and routed before it reaches the model. Define pipelines and agent workflows where retrieval, memory, tools, and generation are transparent and traceable.

**Unified pipelines, sync and async**  

> **Native Async Support**  

Reviewer: ```suggestion
**Native Async Support**  
```

One `Pipeline` runs synchronously or asynchronously and streams token by token. `Agent` can run concurrent tool calls.  

**Modular and customizable**  
Use built-in components for retrieval, indexing, tool calling, memory, and evaluation, or create your own. Add loops, branches, and conditional logic to precisely control how context moves through your pipelines and agent workflows.

**Model- and vendor-agnostic**  

> Integrate with OpenAI, Mistral, Anthropic, Cohere, Hugging Face, Azure OpenAI, AWS Bedrock, local models, and many others. Swap models or infrastructure components without rewriting your system.

Author: ```suggestion
Integrate with OpenAI, Mistral, Anthropic, Cohere, Hugging Face, Google, Azure OpenAI, AWS Bedrock, local models, and many others. Swap models or infrastructure components without rewriting your system.
```
